roles/download/tasks/kubeadm_images.yml

```diff
@@ -4,5 +4,21 @@
     src: "kubeadm-images.yaml.j2"
     dest: "{{ kube_config_dir }}/kubeadm-images.yaml"
 
+- name: kubeadm | Copy kubeadm binary from download dir
```

Comment:
With this addition, The same tasks in roles/kubernetes/node/tasks/install.yml should be removed

Reply:
This task only runs on master nodes, you'll need kubeadm on the worker nodes too in order to join the cluster. As it is using the synchronize module, I thought it wouldn't hurt to not change download role and run it on all nodes anyway. But I can add a condition on the download role to install it only on the worker nodes if you prefer this.

Reply:
added in 3dbf9debf0c6910e12aebc03eef1838009f04dcb